buffer/src/main/java/io/netty/buffer/AdvancedLeakAwareCompositeByteBuf.java

```diff
@@ -1038,23 +1039,29 @@ public CompositeByteBuf touch(Object hint) {
 
     @Override
     public boolean release() {
-        boolean deallocated = super.release();
-        if (deallocated) {
-            leak.close();
-        } else {
-            leak.record();
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by ByteBuf.
+        ByteBuf unwrapped = unwrap();
+        if (super.release()) {
+            boolean closed = ResourceLeakDetector.close(leak, unwrapped);
+            assert closed;
+            return true;
         }
-        return deallocated;
+        leak.record();
+        return false;
     }
 
     @Override
     public boolean release(int decrement) {
-        boolean deallocated = super.release(decrement);
-        if (deallocated) {
-            leak.close();
-        } else {
-            leak.record();
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by ByteBuf.
+        ByteBuf unwrapped = unwrap();
+        if (super.release(decrement)) {
+            boolean closed = ResourceLeakDetector.close(leak, unwrapped);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)